src/Controller/Component/SecurityComponent.php

```diff
@@ -486,6 +490,66 @@ protected function _sortedUnlocked($data)
     }
 
     /**
+     * Get the optional string
+     *
+     * @param array $data Data array
+     * @return array Optional fields and the values they must have if present
+     */
+    protected function _optional(array $data)
+    {
+        if (!empty($data['_Token']['optional'])) {
+            $optional = urldecode($data['_Token']['optional']);
+            $optionalFields = [];
+            if (!empty($optional)) {
+                $optional = explode('|', $optional);
+                foreach ($optional as $field) {
+                    list($name, $hash) = explode('=', $field);
```

Comment:
Yeah, that should simplify things. I was sticking with the | delimiting that the other params used, but there's no compelling reason to do so.